app-next/app/movies/page.jsx

```diff
@@ -147,7 +147,8 @@ export default function ExplorePage() {
       }
     } catch (error) {
       console.error("Error updating favorites:", error);
-      alert("Failed to update favorites. Please try again.");
+      window.location.href = "/login";
```

Comment:
Can you check the error and only redirect if it's due to authentication, otherwise show the alert.

Reply:
I think you should gray out the button if the user is not logged in, instead of redirecting when the click on the button.

Reply:
You can check for the user somewhere earlier on this page and disable the button of the user does not exist, like here: https://github.com/l0rians/groupApp/pull/32#discussion_r1936412126